app/src/main/java/unipd/se18/MignifyResolver.java

```diff
@@ -0,0 +1,111 @@
+package unipd.se18.eanresolvemodule;
+
+import android.util.Log;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * This class implements the Mignify API for EAN13 code resolve
+ * More at https://market.mashape.com/mignify/product-intelligence#
+ * @author Elia Bedin
+ */
+public class MignifyResolver implements EAN {
+
+    //The string resolved from an EAN product code
+    private String product = "";
+    //The string passed to the get method, postponed by the EAN string code
+    private static final String URL =
+            "https://mignify.p.mashape.com/gtins/v1.0/productsToGtin?gtin=";
+    //The authentication string you get from
+    //https://market.mashape.com/mignify/product-intelligence#
+    //after you subscribe to the Mignify API
+    private static final String MASHAPE_AUTH = "Nx432kOz2Wmsh83IAjUv6cUaxT7Rp1tDUbKjsnu7ueSdRFdOPK";
+    //Error text when connecting to the Mashape API
+    private static final String HTTP_ERROR = "Unable to connect to the server.";
+    //Error text if there's no product with entered EAN code in the database
+    private static final String NO_PRODUCT_ERROR =
+            "No product found in the database for that barcode.";
+    //Error text if there was an error parsing the JSON response
+    private static final String JSON_ERROR = "Error parsing the response from the server.";
+
+    /**
+     *
+     * @param EANCode string extracted from an EAN13 barcode
+     * @return the product relative to the input string
+     */
+    @Override
+    public String decodeEAN(String EANCode) {
+
+        //Adding to the URL request the EAN code taken from the barcode
+        String fullURL = URL + EANCode;
+        //HTTP response from get method to Mignify API
+        HttpResponse<JsonNode> response = null;
+        //Unirest library useful for making an HTTP request for the Mashape API
+        //More at http://unirest.io/java
+        try {
+            response = Unirest
```

Comment:
Reading the Unirest documentation for java, I saw that this kind of request blocks the application; have you considered doing an asynchronous request?
You can see some examples here, always with Unirest:
https://www.programcreek.com/java-api-examples/index.php?api=com.mashape.unirest.http.async.Callback